Apps/Src/Temperature.c

```diff
@@ -155,15 +163,24 @@ ErrorStatus Temperature_ChannelConfig(uint8_t tempChannel) {
  * @param mV from ADC
  * @return temperature in Celsius (Fixed Point with .001 resolution) 
  */
-int milliVoltToCelsius(float milliVolt){
+int milliVoltToCelsius(uint32_t milliVolt){
+	// Adapted from: 
+	/*
 	// Typecasting to get rid of warnings
 	float sumInRt = (float)(-13.582)*(float)(-13.582) + (float)4.0 * (float)0.00433 * ((float)2230.8 - milliVolt);
 	float rt = sqrt(sumInRt);				
 	float numerator = (float)13.582 - rt;
 	float denom = (2.0 * - 0.00433);		
 	float frac = numerator/denom;			
-	float retVal = frac + 30;			
-	
+	float retVal = frac + 30;	
+	*/
+
+	uint32_t sumInRt = (184471 + 17 * (2230800 - milliVolt * 1000)) / 1000;
+	uint32_t rt = uintSqrt(sumInRt);
+	int32_t numerator = 13582000 - rt * 1000000;
+	int32_t denominator = -8660;
+	int32_t retVal = numerator / denominator + 30;
```

Comment:
Might be good to get some idea of how much precision we're losing from this, and whether there's any other order of operations that would buy us more precision.

Reply:
I replaced this with a lookup table because I think we have the flash space, and I don't want to worry about losing precision